src/Avalonia.Input/InputElement.cs

```diff
@@ -343,6 +352,15 @@ public bool IsEnabled
             get { return GetValue(CursorProperty); }
             set { SetValue(CursorProperty, value); }
         }
+        
+        /// <summary>
+        /// Gets a value indicating whether keyboard focus is anywhere within the element or its visual tree child elements.
+        /// </summary>
+        public bool IsKeyboardFocusWithin
```

Comment:
I know this naming is from WPF, but should it be limited to Keyboard only? I mean, it also could be possible to use controllers and other kind of input apart from classic keyboard/mouse/touch.

Reply:
Constraining the term to KeyboardFocus may be necessary here to disambiguate from other focus types -- although not sure if Avalonia supports LogicalFocus concepts.

Reply:
We have "IsFocused" property, so this one could be "IsFocusedWithin".
But "IsKeyboardFocusWithin" should be also considered since it brings compatibility with WPF.

Reply:
We decided to keep the WPF naming.